google/cloud/security/scanner/scanners/bigquery_scanner.py

```diff
@@ -24,28 +28,70 @@
 class BigqueryScanner(base_scanner.BaseScanner):
     """Pipeline to pull Big Query acls data from DAO"""
 
-    def __init__(self, snapshot_timestamp):
+    def __init__(self, global_configs, scanner_configs, snapshot_timestamp,
+                 rules):
         """Initialization.
 
         Args:
             snapshot_timestamp (str): The snapshot timestamp
         """
         super(BigqueryScanner, self).__init__(
-            snapshot_timestamp)
-        self.snapshot_timestamp = snapshot_timestamp
+            global_configs,
+            scanner_configs,
+            snapshot_timestamp,
+            rules)
+        self.rules_engine = bigquery_rules_engine.BigqueryRulesEngine(
+            rules_file_path=self.rules,
+            snapshot_timestamp=self.snapshot_timestamp)
+        self.rules_engine.build_rule_book(self.global_configs)
+
+    def _flatten_violations(self, violations):
```

Comment:
Nice! I like it. I wonder if we can make this to be violation agnostic and have a single flattening function for all types of violations in the future.